src/Backend/Core/Engine/Meta.php

```diff
@@ -625,4 +570,14 @@ public function getData()
     {
         return $this->data;
     }
+
+    /**
+     * @return MetaEntity
+     */
+    public function getMetaEntity()
+    {
+        $this->validate();
+
+        return MetaEntity::fromBackendMeta($this);
```

Comment:
Wouldn't it be better to create a factory for this?
$this->commonMetaFactory->createFromBackend($this);
Which then uses ->create() of the MetaEntity